src/main/java/us/kbase/auth2/lib/UserName.java

```diff
@@ -35,8 +39,8 @@ public class UserName extends Name {
 		}
 	}
 	
-	private static final String INVALID_CHARS_REGEX = "[^a-z\\d_]+";
-	private final static Pattern INVALID_CHARS = Pattern.compile(INVALID_CHARS_REGEX);
+	private static final Pattern FORCE_ALPHA_FIRST_CHAR = Pattern.compile("^[^a-z]+");
```

Comment:
I hope 2pac never made a KBase account!

Reply:
I'm not sure how deep KBase has penetrated into the gangsta rap community

Reply:
Something to check with KBase outreach about.